only-allow: not found / MODULE_NOT_FOUND

Opening the issue, not because I am convinced that the source of the problem is here, but just to let you guys know that there is such issue, and possible fix.

**Describe the bug**
- we are using `optimism-sdk` as dependency
- locally, all seem to work
- on circleCI or on heroku, the `yarn install` command (seemingly randomly) fails with

```
Exit code: 127
Command: npx only-allow pnpm
Arguments: 
Directory: /home/circleci/repo/node_modules/@eth-optimism/core-utils
Output:
npm WARN exec The following package was not found and will be installed: only-allow@1.2.1
sh: 1: only-allow: not found
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

Exited with code exit status 127
```

or 

```
error /home/circleci/repo/node_modules/@eth-optimism/contracts-bedrock: Command failed.
Exit code: 1
Command: npx only-allow pnpm
Arguments: 
Directory: /home/circleci/repo/node_modules/@eth-optimism/contracts-bedrock
Output:
npm WARN exec The following package was not found and will be installed: only-allow@1.2.1
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module '/home/circleci/.npm/_npx/0b83cd9ca5e1325c/node_modules/.bin/only-allow'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.17.1
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```


**To Reproduce**
 - hard to do, for us, the `yarn install` script was failing randomly, we haven't figured out what is it dependent on

**Expected behavior**
- yarn install works, only allow gets installed and is found

**System Specs:**
"@eth-optimism/sdk": "^3.1.6"

**Additional context**
Possibly related
https://github.com/pnpm/only-allow/issues/11
https://github.com/npm/cli/issues/2660
https://github.com/Lightprotocol/light-protocol/issues/381
https://github.com/juliencrn/usehooks-ts/issues/420

**Possible fix**
- we added `only-allow` package as `devDependency` which seemed to fix the issue.

--- a/.changeset/calm-boxes-do.md
+++ b/.changeset/calm-boxes-do.md
@@ -2,4 +2,4 @@
 '@eth-optimism/sdk': patch
 ---
 
-fixed issue with vercel builds failing on preinstall command
+Fixed an issue where Vercel builds were failing due to the `preinstall` command.

--- a/.changeset/calm-boxes-do.md
+++ b/.changeset/calm-boxes-do.md
@@ -1,0 +1,5 @@
+---
+'@eth-optimism/sdk': patch
+---
+
+fixed issue with vercel builds failing on preinstall command

--- a/packages/chain-mon/package.json
+++ b/packages/chain-mon/package.json
@@ -29,7 +29,6 @@
     "test:coverage": "nyc hardhat test && nyc merge .nyc_output coverage.json",
     "build": "tsc -p ./tsconfig.json",
     "clean": "rimraf  dist/ ./tsconfig.tsbuildinfo",
-    "preinstall": "npx only-allow pnpm",
     "lint": "pnpm lint:fix && pnpm lint:check",
     "pre-commit": "lint-staged",
     "lint:fix": "pnpm lint:check --fix",

--- a/packages/common-ts/package.json
+++ b/packages/common-ts/package.json
@@ -12,7 +12,6 @@
     "all": "pnpm clean && pnpm build && pnpm test && pnpm lint:fix && pnpm lint",
     "build": "tsc -p tsconfig.json",
     "clean": "rimraf dist/ ./tsconfig.tsbuildinfo",
-    "preinstall": "npx only-allow pnpm",
     "lint:check": "eslint . --max-warnings=0",
     "lint:fix": "pnpm lint:check --fix",
     "lint": "pnpm lint:fix && pnpm lint:check",

--- a/packages/contracts-bedrock/package.json
+++ b/packages/contracts-bedrock/package.json
@@ -34,7 +34,6 @@
     "validate-spacers:no-build": "npx tsx scripts/checks/check-spacers.ts",
     "validate-spacers": "pnpm build && pnpm validate-spacers:no-build",
     "clean": "rm -rf ./artifacts ./forge-artifacts ./cache ./tsconfig.tsbuildinfo ./tsconfig.build.tsbuildinfo ./scripts/go-ffi/go-ffi ./.testdata ./deployments/hardhat/*",
-    "preinstall": "npx only-allow pnpm",
     "pre-pr:no-build": "pnpm gas-snapshot:no-build && pnpm snapshots && pnpm semver-lock && pnpm autogen:invariant-docs && pnpm lint && pnpm bindings:go:no-build",
     "pre-pr": "pnpm clean && pnpm build:go-ffi && pnpm build && pnpm pre-pr:no-build",
     "pre-pr:full": "pnpm test && pnpm validate-deploy-configs && pnpm validate-spacers && pnpm pre-pr",

--- a/packages/contracts-ts/package.json
+++ b/packages/contracts-ts/package.json
@@ -37,7 +37,6 @@
   "scripts": {
     "build": "tsup",
     "clean": "rm -rf ./dist",
-    "preinstall": "npx only-allow pnpm",
     "generate": "wagmi generate && pnpm build && pnpm lint:fix",
     "generate:check": "pnpm generate && git diff --exit-code ./addresses.json && git diff --exit-code ./abis.json",
     "lint": "prettier --check .",

--- a/packages/core-utils/package.json
+++ b/packages/core-utils/package.json
@@ -11,7 +11,6 @@
     "all": "pnpm clean && pnpm build && pnpm test && pnpm lint:fix && pnpm lint",
     "build": "tsc -p tsconfig.json",
     "clean": "rimraf dist/ ./tsconfig.tsbuildinfo",
-    "preinstall": "npx only-allow pnpm",
     "lint": "pnpm lint:fix && pnpm lint:check",
     "lint:check": "eslint . --max-warnings=0",
     "lint:fix": "pnpm lint:check --fix",

--- a/packages/sdk/package.json
+++ b/packages/sdk/package.json
@@ -12,7 +12,6 @@
     "all": "pnpm clean && pnpm build && pnpm test && pnpm lint:fix && pnpm lint",
     "build": "tsc -p tsconfig.json",
     "clean": "rimraf dist/ ./tsconfig.tsbuildinfo",
-    "preinstall": "npx only-allow pnpm",
     "lint": "pnpm lint:fix && pnpm lint:check",
     "lint:check": "eslint . --max-warnings=0",
     "lint:fix": "pnpm lint:check --fix",

